projects/storefrontlib/src/shared/components/item-counter/item-counter.component.ts

```diff
@@ -1,219 +1,118 @@
+   * the minimum. The zero value can be used to remove an item.
    */
-  increment(): void {
-    this.manualChange(this.value + this.step);
-    this.setFocus(true);
-  }
+  @Input() allowZero: false;
+
+  private _control$: Observable<FormControl>;
 
   /**
-   * Verify value that it can be decremented, if yes it does that.
+   * In readonly mode the item counter will only be shown as a label,
+   * the form controls are not rendered.
+   * Please not that readonly is different from the `disabled` form state.
+   * @default false
    */
-  decrement(): void {
-    this.manualChange(this.value - this.step);
-    this.setFocus(false);
-  }
+  @HostBinding('class.readonly') @Input() readonly = false;
 
-  // ControlValueAccessor interface
+  @ViewChild('qty', { static: false }) private input: ElementRef<
+    HTMLInputElement
+  >;
 
-  registerOnTouched(fn: Function): void {
-    this.onTouch = fn;
+  @HostListener('click')
+  handleClick() {
+    this.input.nativeElement.focus();
   }
 
-  registerOnChange(fn: Function): void {
-    this.onModelChange = fn;
+  increment() {
+    // it's too early to use the `stepUp` and `stepDown` API...
+    // let's wait for FF: https://caniuse.com/#search=stepUp
+    this.control.setValue(this.control.value + this.step);
+    this.control.markAsDirty();
   }
 
-  writeValue(value: number): void {
-    this.value = value || this.min || 0;
-    this.onModelChange(this.value);
+  decrement() {
+    this.control.setValue(this.control.value - this.step);
+    this.control.markAsDirty();
   }
 
   /**
-   * Set up new value for input and emit event outside
+   * Returns an observable with the control. The value changes of the
+   * control are intercepted in order to suppress invalid values.
    */
-  updateValue(updatedQuantity: number): void {
-    if (!this.async) {
-      // If the async flag is true, then the parent component is responsible for updating the form
-      this.writeValue(updatedQuantity);
+  getControl(): Observable<FormControl> {
+    if (!this._control$) {
+      this._control$ = this.control.valueChanges.pipe(
+        startWith(this.control.value),
+        tap(value =>
+          this.control.setValue(this.getValidCount(value), { emitEvent: false })
+        ),
+        map(() => this.control)
+      );
     }
-
-    // Additionally, we emit a change event, so that users may optionally do something on change
-    this.update.emit(updatedQuantity);
-    this.onTouch();
+    return this._control$;
   }
 
-  /**
-   * Determines which HTML element should have focus at a given time
-   */
-  setFocus(isIncremented: boolean): void {
-    if (this.isMaxOrMinValueOrBeyond()) {
-      this.input.nativeElement.focus();
-    } else if (isIncremented) {
-      this.incrementBtn.nativeElement.focus();
-    } else {
-      this.decrementBtn.nativeElement.focus();
+  private getValidCount(value: number) {
+    let newValue = Number(value);
```

Comment:
```suggestion
    let newValue = value;
```

Is it necessary?

Reply:
Nope, not needed anymore since we moved to numeric input fields.